### PR TITLE
Create `ApplicationPasswordUseCase` protocol.

### LINKED ISSUE #8381: Create Application password protocol
To start the application password authentication for Woo REST API work, we need to add a protocol to interface the work of generating and deleting application passwords.

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #8381

### Description
This PR introduces `ApplicationPasswordUseCase` protocol to act as an interface for application password generation and deletion. A dummy implementation of this protocol will be used in future PR to work parallelly. 

### Testing instructions
CI passing is enough.

### Screenshots
NA

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Authentication/Application password/ApplicationPasswordUseCase.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import Foundation
+import WordPressShared
+
+struct ApplicationPassword {
+    /// WordPress org username that the application password belongs to
+    ///
+    let wpOrgUername: String
```
Comment (itsmeichigo) [inline WooCommerce/Classes/Authentication/Application password/ApplicationPasswordUseCase.swift]: Typo 😅 
```suggestion
    let wpOrgUsername: String
```

**Review by itsmeichigo — APPROVED** [review]
🚢

**Comment by itsmeichigo** [comment]
One thing that I missed: I thought `ApplicationPasswordUseCase` would be in the `Networking` layer since we planned to let the `AlamofireNetwork` work with it. Should we move it there and import the use case class on the UI layer later if needed?

**Comment by selanthiraiyan** [comment]
> One thing that I missed: I thought `ApplicationPasswordUseCase` would be in the `Networking` layer since we planned to let the `AlamofireNetwork` work with it. Should we move it there and import the use case class on the UI layer later if needed?

I didn't think about this. I agree that it should reside in the `Networking` layer. Moved in https://github.com/woocommerce/woocommerce-ios/pull/8382/commits/22280d343ea8951382bd9ae6cbc3d03c0563f961

**Pull request merged**